app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -448,23 +448,26 @@ object AppRepository {
         }
         val dayRanges = loadSessionsForAllDays(includeEngelsystemShifts = false)
                 .toDayRanges()
+        val oldShifts = loadEngelsystemShiftsForAllDays()
         val sessionizedShifts = shifts
                 .also { logging.d(LOG_TAG, "Shifts unfiltered = ${it.size}") }
                 .cropToDayRangesExtent(dayRanges)
                 .also { logging.d(LOG_TAG, "Shifts filtered = ${it.size}") }
                 .toSessionAppModels(logging, ENGELSYSTEM_ROOM_NAME, dayRanges)
                 .sanitize()
-        val toBeUpdatedSessions = loadSessionsForAllDays(false) // Drop all shifts before ...
+        val shiftChanges = computeSessionsWithChangeFlags(sessionizedShifts, oldShifts)
+        if (oldShifts.isEmpty() || shiftChanges.foundAnyChanges) {
```

Comment:
If `Session` implements a proper `equals()` method this could simply be

```kotlin
if (oldShifts != sessionizedShifts) {
   /// …
}
```

…and no changes to `ScheduleChanges` should be necessary.

Reply:
True. Migrating the `Session` class is on my list. Meanwhile, the comparison has to happen here.